File: readinglvl.c
// #include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdbool.h>
#include <math.h>

int main(void)
{

    // TODO: redo without cs50 get_string, or download the .h

    // string text;
    int letters = 0;
    int words = 1;
    int sentences = 0;

    // user input//
    // text = get_string("Type in your text: ");

    // count letters//
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
            letters++;
    }
    // printf("total letters is %d\n", letters);//

    // count words//
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
            words++;
    }
    // printf("total words is %d\n", words);//

    // count sentences//
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
            sentences++;
    }
    // printf("total sentences are: %d\n", sentences);//

    // equation//
    float calc = (0.0588 * letters / words * 100) - (0.296 * sentences / words * 100) - 15.8;
    int index = round(calc);

    // if statement to print reading level//
    if (index < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    else
    {
        printf("Grade %d\n", index);
    }
}

File: readinglvl.py
import re


def main():
    # Variables
    letters = 0
    words = 1
    sentences = 0

    # Input
    text = input("Input your text: ")
    # if not text.isalpha():
    #     print("Please enter only alphabetical characters for your input.")

    # Count letters
    letters = len(text) - text.count(" ") - text.count(".") - text.count("!") - text.count("?")
    # for l in range(len(text)):
    #     if text[l].isalpha():
    #         letters += 1

    # Count words
    # words = len(re.findall(r"\w+", text))
    for i in range(len(text)):
        if (i == 0 and text[i] != " ") or (i != len(text) - 1 and text[i] == " " and text[i + 1] != " "):
            words += 1

    # Count Sentences
    for s in text:
        if s.count("."):
            sentences += 1
        if s.count("?"):
            sentences += 1
        if s.count("!"):
            sentences += 1

    # Equation + new variables for equation
    L = letters / words * 100
    S = sentences / words * 100
    # cl_index = 0.0588 * L - 0.296 * S - 15.8
    index = round(0.0588 * L - 0.296 * S - 15.8)

    # Print checks
    # print(f"letters: {letters}")
    # print(f"words: {words}")
    # print(f"sentences: {sentences}")

    # Print grade statements
    if index < 1:
        print("Grade: Before Grade 1.")
    elif index >= 16:
        print("Grade: Grade 16+")
    else:
        print(f"Grade: {index}")



if __name__ == "__main__":
    main()